### PR TITLE
To start behaviourspace run with view update and update plots and monitor disabled

### PR DESCRIPTION
**Pull request by diparthshah**
This provides fix for issue : #1376

### CODE REVIEW

**Comment by mrerrormessage** [comment]
@diparthshah , thanks for the PR. This needs some more work before it's ready to review. #1376 isn't about starting each behaviorspace run with the view, plots, and monitors disabled, but about giving the user the option to disable them before the run starts. If you haven't already, try working through the [user manual's Behaviorspace example](https://ccl.northwestern.edu/netlogo/docs/behaviorspace.html) to get an idea of how behaviorspace is used. #1376 references the RunOptionsDialog. This is a Java class, and when it's being used it looks like this:

screen shot 2018-02-16 at 11 32 45 am

The task in #1376 is to add two additional checkboxes to this (one for plots/monitors disabled, one for view disabled) and then make use of those options in `Supervisor.start()` as the world is being configured. The options selected should also be passed into `ProgressDialog` so that it reflects the accurate state of the run. This is a fairly sizable task, so please contact me if you have questions.

**Comment by diparthshah** [comment]
@mrerrormessage 
i have added the new check box in the Run Options dialog , i needed help regarding to how i could pass the status from Run Options dialog window to Progress dialog window. I have tried few times but didn't got expected results. 
plots

**Comment by diparthshah** [comment]
i have added a procedure in ProgressDialog which changes value of boolean variable.
Procedure is called in Supervisor , if the checkbox is ticked it disables the update.
i am pasting piece of code to clear the picture

ProgressDialog
``
private var updateView = true

def updateViewStatus(checked:Boolean): Unit ={
    if(checked){
      updateView = false
      System.out.println("refresh status "+updateView)
    }
    else {
      updateView = true
      System.out.println("refresh status "+updateView)
    }
  }

val displaySwitch = new JCheckBox(displaySwitchAction)
    displaySwitch.setSelected(updateView)
    workspace.displaySwitchOn(updateView)

**Comment by mrerrormessage** [comment]
Haven't had a chance to look at this yet, but I will review this tomorrow and get feedback to you. Thanks!

**Comment by diparthshah** [comment]
I have not pushed above mentioned changes , just sharing my idea of doing that task. Will wait for your opinion. 
-Thank you

File: netlogo-gui/src/main/lab/gui/RunOptionsDialog.scala [hunk 1]
```diff
@@ -16,9 +16,14 @@ class RunOptionsDialog(parent: java.awt.Dialog,
     private val prefs = Preferences.userNodeForPackage(RunOptionsDialog.this.getClass)
     def spreadsheet = prefs.getBoolean("spreadsheet", true)
     def table = prefs.getBoolean("table", false)
+    /* add two more boolean property for disabling view,plots and monitor update */
+    def update_view = prefs.getBoolean("update_view",false)
+    def update_plots_monitor = prefs.getBoolean("update_plots_monitor",false)
```
Comment (mrerrormessage) [inline netlogo-gui/src/main/lab/gui/RunOptionsDialog.scala]: In Java/Scala, variable names (and symbolic names like the string names here) are `camelCased` and not `snake_cased`, as here.

File: netlogo-gui/src/main/lab/gui/Supervisor.scala [hunk 2]
```diff
@@ -89,6 +90,13 @@ class Supervisor(dialog: java.awt.Dialog,
         worker.protocol,
         new java.io.PrintWriter(new java.io.FileWriter(fileName))))
     }
+    /* if condition for newly added options (view,plots and monitor)*/
+    if(options.update_view){
+      System.out.println("Disable View Update")
+    }
+    if(options.update_plots_monitor){
+      System.out.println("Disable Plots and Monitor Update")
+    }
```
Comment (mrerrormessage) [inline netlogo-gui/src/main/lab/gui/Supervisor.scala]: I assume this is debugging code?

File: netlogo-gui/src/main/lab/gui/Supervisor.scala [hunk 3]
```diff
@@ -14,7 +14,8 @@ import org.nlogo.nvm.LabInterface.ProgressListener
 import org.nlogo.api.LogoException
 
 object Supervisor {
-  case class RunOptions(threadCount: Int, table: Boolean, spreadsheet: Boolean)
+            /* add argument for newly added boolean values(view,plots and monitor) */
+  case class RunOptions(threadCount: Int, table: Boolean, spreadsheet: Boolean,update_view: Boolean,update_plots_monitor: Boolean)
```
Comment (mrerrormessage) [inline netlogo-gui/src/main/lab/gui/Supervisor.scala]: Remove comment, use `camelCase`

**Comment by diparthshah** [comment]
Thank you sir for your suggestion. 
I will make changes as you suggested and push them, also will squash changes into single commit for the PR thing.

File: netlogo-gui/src/main/lab/gui/RunOptionsDialog.scala [hunk 4]
```diff
@@ -32,16 +36,20 @@ class RunOptionsDialog(parent: java.awt.Dialog,
...
+    var updateView = Prefs.updateView
+    var updatePlotsAndMonitor= Prefs.updatePlotsAndMonitor
     var threadCount = Runtime.getRuntime.availableProcessors
     val classDisplayName = "Run options"
     val propertySet =
       List(
         Property("spreadsheet", Property.Boolean, "Spreadsheet output"),
         Property("table", Property.Boolean, "Table output"),
+        Property("updateView",Property.Boolean,"Update View"),
+        Property("updatePlotsAndMonitor",Property.Boolean,"Update Plots and Monitor"),
```
Comment (nicolaspayette) [inline netlogo-gui/src/main/lab/gui/RunOptionsDialog.scala]: The user facing strings should only have the first letter capitalized and (more importantly) "monitor" should be plural, i.e.: `"Update view"` and `"Update plots and monitors"`. "Monitor" should be plural in property and variable names as well.

**Pull request closed without merging**